lib/rbnacl/hash/blake2b.rb

```diff
@@ -64,19 +76,61 @@ def initialize(opts = {})
 
         @salt     = opts.fetch(:salt, EMPTY_SALT)
         @salt     = Util.zero_pad(SALTBYTES, @salt)
+        @incycle  = false
+        @instate  = nil
       end
 
       # Calculate a Blake2b digest
       #
       # @param [String] message Message to be hashed
       #
       # @return [String] Blake2b digest of the string as raw bytes
-      def digest(message)
+      def digest(*message)
+        return final_digest if message.empty?
```

Comment:
Ugh, I don't like this. I'd rather this method not have two responsibilities. I'd suggest just renaming `final_digest` to `digest`, and making the rest of this method a completely separate class method.